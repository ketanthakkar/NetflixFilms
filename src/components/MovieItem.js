import React from 'react';
import PropTypes from 'prop-types';
import { Link } from "react-router-dom";

const MovieItem = ({movieItem}) => {
    return (
        <section className="movieitem-container">
            <figure>
                <Link to={`/film/${movieItem.id}`}>
                    <img className="movieitem-img" src={movieItem.poster_path} alt={movieItem.tagline}/>
                </Link>
                <figcaption className="movieitem-detail">
                    <span>{movieItem.title.toUpperCase()}</span>
                    <span className="movieitem-year">{movieItem.release_date}</span>
                </figcaption>
                <span className="movieitem-genres">{movieItem.genres.join(" & ")}</span>
            </figure>
        </section>
    )
}

MovieItem.propTypes = {
    movieItem: PropTypes.object
};

export default MovieItem;