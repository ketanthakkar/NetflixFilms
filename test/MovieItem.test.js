import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter as Router } from 'react-router-dom';
import MovieItem from '../src/components/MovieItem';

describe('should renders MovieItem correctly', () => {
  const movieData = {
    id: 337167,
    title: 'Fifty Shades Freed',
    tagline: "Don't miss the climax",
    vote_average: 6.1,
    vote_count: 1195,
    release_date: '2018-02-07',
    poster_path: 'https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg',
    overview: 'Believing they have left behind shadowy figures from their past',
    budget: 55000000,
    revenue: 136906000,
    genres: [
      'Drama',
      'Romance',
    ],
    runtime: 106,
  };
  const tree = mount(<Router keyLength={0}><MovieItem key={ movieData.id } movieItem={ movieData }/></Router>);
  test('Snapshot test', () => {
    expect(tree).toMatchSnapshot();
  });
});
