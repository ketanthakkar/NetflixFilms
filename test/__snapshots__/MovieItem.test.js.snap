// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders MovieItem correctly Snapshot test 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "2jqxhw",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "2jqxhw",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <MovieItem
      key="337167"
      movieItem={
        Object {
          "budget": 55000000,
          "genres": Array [
            "Drama",
            "Romance",
          ],
          "id": 337167,
          "overview": "Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.",
          "poster_path": "https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg",
          "release_date": "2018-02-07",
          "revenue": 136906000,
          "runtime": 106,
          "tagline": "Don't miss the climax",
          "title": "Fifty Shades Freed",
          "vote_average": 6.1,
          "vote_count": 1195,
        }
      }
    >
      <section
        className="movieitem-container"
      >
        <figure>
          <Link
            replace={false}
            to="/film/337167"
          >
            <a
              href="/film/337167"
              onClick={[Function]}
            >
              <img
                alt="Don't miss the climax"
                className="movieitem-img"
                src="https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg"
              />
            </a>
          </Link>
          <figcaption
            className="movieitem-detail"
          >
            <span>
              FIFTY SHADES FREED
            </span>
            <span
              className="movieitem-year"
            >
              2018-02-07
            </span>
          </figcaption>
          <span
            className="movieitem-genres"
          >
            Drama & Romance
          </span>
        </figure>
      </section>
    </MovieItem>
  </Router>
</MemoryRouter>
`;
